EXTRA_FIM/pyiron_job.py

```diff
@@ -0,0 +1,247 @@
+            e_fermi,
+            ionization_energies=self.input["ionization_energies"],
+        )
+        self.input["total_kpoints"] = waves_reader.nk
+        self.input["simulator_dict"] = sim
+        self.input["z_max"] = sim["z_max"]  # rename later
+        self.input["izstart_min"] = sim["izstart_min"]  # rename later
+        self.input["izend"] = sim["izend"]  # rename later
+        self.input["limit"] = sim["limit"]  # rename later
+        self.input["cutoff"] = sim["cutoff"]  # rename later
+        self.input["E_fermi"] = sim["E_fermi"]  # rename later
+        self.input["E_max"] = sim["E_max"]  # rename later
+        return sim
+
+    def run_static(self):
+
+        self.project_hdf5.create_working_directory()
+        # self.suggest_input_dict
+        pot_ext, elec_ext = self.extrpolate_potential()
+        waves_reader = sx_nc_waves_reader(self.input["waves_directory"] + "/waves.sxb")
+        fimsim = fim.FIM_simulations(
+            self.input["simulator_dict"],
+            reader=waves_reader,
+            V_total=pot_ext,
+            V_elstat=elec_ext,
+        )
+        self.status.submitted = True
+        fimsim.sum_single_k(self.input["kpoint"], path=self.working_directory)
+        self.status.finished = True
+
+
+class ExtraFimSimulatorJobGenerator(JobGenerator):
+    """Job generator class for extra fim simulator pyiron jobs"""
+
+    @property
+    def parameter_list(self):
+        """
+
+        Returns:
+            (list)
+        """
+        parameter_lst = []
+        kpoints = self._master.input.get("kpoint")
+        if kpoints is None:
+            waves_reader = sx_nc_waves_reader(
+                Path(self._master.input["waves_directory"]) / "waves.sxb"
+            )
+        for k in range(waves_reader.nk):
+            parameter_lst.append(k)
+        return parameter_lst
+
+    def job_name(self, parameter):
+        k_point = parameter
+        return f"{self._master.job_name}_kpoint_{k_point}"
+
+    def modify_job(self, job, parameter):
+        job.structure = self._master.get_structure
+        for k in self._master.input.keys():
+            job.input[k] = self._master.input[k]
+        job.input.kpoint = parameter
+        return job
+
+
+class ExtraFimSimulator(ParallelMaster):
+    """ "Pyiron Extra FIM simulator job class to make subjobs for each k point"""
+
+    def __init__(self, project, job_name):
+        super(ExtraFimSimulator, self).__init__(project, job_name)
+        self.__version__ = "0.1.0"
+        self.input["waves_directory"] = None
+        self.input["waves_reader"] = None
+        self.input["kpoint"] = None
+        self.input["ionization_energies"] = None
+        self.input["extrapolate_potential"] = False
+        self.input["total_kpoints"] = None
+        self._job_generator = ExtraFimSimulatorJobGenerator(self)
+        self.ref_job = ExtraFimSimulatorRefJob(project=project, job_name=job_name)
+
+    def extrpolate_potential(self):
```

Comment:
typo in function name
```suggestion
    def extrapolate_potential(self):
```